catalog/src/test/java/it/eng/catalog/rest/api/CatalogAPIControllerTest.java

```diff
@@ -0,0 +1,90 @@
+package it.eng.catalog.rest.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import it.eng.catalog.exceptions.CatalogNotFoundAPIException;
+import it.eng.catalog.service.CatalogService;
+import it.eng.catalog.util.MockObjectUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class CatalogAPIControllerTest {
+
+
+    CatalogAPIController catalogAPIController;
+    @Mock
+    private CatalogService catalogService;
+
+    @BeforeEach
```

Comment:
We should choose this or @InjectMocks (negotiation module)
and be consistent in all junit tests

Reply:
Noted that - @InjectMocks is used!